### DISCUSSION TITLE
finding the repo for a formula

### DISCUSSION THREAD

**Discussion by beej** [Homebrew/discussions#757] [Everyday usage]
Hello helpful internet denizens 👋🏼

A formula gave me an error that appears to me to be due to the author(s)' missing an edge case.   I see that https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md has:
Open an issue on the formula's repository or on Homebrew/brew if it's not a formula-specific issue.

I believe this to be a formula-specific issue.  How do I go about finding the repository for it?

In this case, the formula is for gnupg, but I'm just as interested in the generic case as I am in the specific case.

Thanks!

**Comment by SMillerDev** [comment]
`brew home` will get you the homepage, which often is or contains the repository.

**Reply by beej** [reply]
Thanks!

That seems like it would be often helpful.  In this case, `brew home gnupg` just opened https://gnupg.org/ in my browser which hasn't gotten me any closer to finding the gnupg formula's repository.

Any more clues?

**Reply by carlocab** [reply]
Do you want to change the gnupg formula, or do you have to change gnupg?

Former: https://github.com/Homebrew/homebrew-core
Latter: https://lists.gnupg.org/mailman/listinfo/gnupg-devel (not a repo, but their dev mailing list)

**Reply by beej** [reply]
`brew info gnupg` has:
> From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gnupg.rb

Does that mean that the formula's repository is the main Homebrew one and I should open an issue there?

**Reply by beej** [reply]
> Do you want to change the gnupg formula, or do you have to change gnupg?

I think I want the gnupg formula to be changed, but I don't know enough about the root cause of the problem to be sure at this point.

> Former: https://github.com/Homebrew/homebrew-core

Thanks!  That's what I was after in my specific case.

The answer in the generic case appears to be:
"Run `brew info [FORMULA]` and look at the final `From: ` line".

Right?

**Reply by SMillerDev** [reply]
There isn't a "main homebrew repository", there is Homebrew/brew for the `brew` tool. There is Homebrew/homebrew-core for the homebrew/core tap containing all software definitions that will be installed automatically when you install homebrew (with Homebrew/linuxbrew-core if you're on linux) and there is Homebrew/cask for all cask definitions.

**Reply by beej** [reply]
I see.

So where would be the proper place to report what I think is an issue with the gnupg software definition which I understand to be part of the homebrow/core tap?

**Reply by beej** [reply]
I think https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md could use some more info related to this discussion to avoid the kind of confusion I ended up in.
